blockchain/filedao/filedao.go

```diff
@@ -84,20 +84,25 @@ type (
 
 // NewFileDAO creates an instance of FileDAO
 func NewFileDAO(cfg db.Config) (FileDAO, error) {
-	header, err := checkMasterChainDBFile(cfg.DbPath)
-	if err == ErrFileInvalid || err == ErrFileCantAccess {
+	header, err := readFileHeader(cfg.DbPath, FileAll)
+	switch err {
```

Comment:
if err != nil {
    if err != ErrFileNotExist {
        return nil, err
    }
    xxx
}

Reply:
updated

Reply:
The original pattern would be better
```
if err ！= nil && err != ErrFileNotExist {
		return nil, err
}

if err == ErrFileNotExist {}
```